src/webapp/reports/csy-audit-operative/csy-audit-operative-list/Filters.tsx

```diff
@@ -0,0 +1,233 @@
+import React, { useMemo, useState } from "react";
+import { OrgUnitsFilterButton } from "../../../components/org-units-filter/OrgUnitsFilterButton";
+import { useAppContext } from "../../../contexts/app-context";
+import { Id } from "../../../../domain/common/entities/Base";
+import styled from "styled-components";
+import { Dropdown, DropdownProps } from "@eyeseetea/d2-ui-components";
+import i18n from "../../../../locales";
+import _ from "lodash";
+import { AuditType } from "../../../../domain/reports/csy-audit-operative/entities/AuditItem";
+
+export interface FiltersProps {
+    values: Filter;
+    options: FilterOptions;
+    onChange: React.Dispatch<React.SetStateAction<Filter>>;
+}
+
+export interface Filter {
+    auditType: AuditType;
+    orgUnitPaths: Id[];
+    periodType: string;
+    year: string;
+    quarter?: string;
+}
+
+export type FilterOptions = {
+    periods: string[];
+};
+
+export const Filters: React.FC<FiltersProps> = React.memo(props => {
+    const { config, api } = useAppContext();
+    const { values: filter, options: filterOptions, onChange } = props;
+
+    const [periodType, setPerType] = useState<string>("yearly");
+    const rootIds = React.useMemo(
+        () =>
+            _(config.currentUser.orgUnits)
+                .map(ou => ou.id)
+                .value(),
+        [config]
+    );
+
+    const periodTypeItems = React.useMemo(() => {
+        return [
+            { value: "yearly", text: i18n.t("Yearly") },
+            { value: "quarterly", text: i18n.t("Quarterly") },
+        ];
+    }, []);
+
+    const yearItems = useMemoOptionsFromStrings(filterOptions.periods);
+
+    const quarterPeriodItems = React.useMemo(() => {
```

Comment:
Not a big deal, but since we're not really doing any modification on these objects we can move them outside the component.

```tsx
const quarterPeriodItems = [
            { value: "Q1", text: i18n.t("Jan - March") },
            { value: "Q2", text: i18n.t("April - June") },
            { value: "Q3", text: i18n.t("July - September") },
            { value: "Q4", text: i18n.t("October - December") },
        ]
        
   function MyComponent() {
     // use quarterPeriodItems directly
     return (
             <SingleDropdownStyled
                        items={quarterPeriodItems}
                        value={filter.quarter}
                        onChange={setQuarterPeriod}
                        label={i18n.t("Quarter")}
                        hideEmpty
                    />
     )
   }

```